### ISSUE TITLE
file.managed diff is switched when using template in salt-ssh 2018.3

### ISSUE THREAD

**Issue by anlutro** [saltstack/salt#46931]
Hopefully these two examples speak for themselves.

`file.managed` using contents:

```
$ cat states/test.sls 
/tmp/test:
  file.managed:
    - contents: new

$ salt-ssh example state.apply test test=true
example:
----------
          ID: /tmp/test
    Function: file.managed
      Result: None
     Comment: The file /tmp/test is set to be changed
     Started: 20:27:56.743931
    Duration: 54.081 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -1 +1 @@
                  -old
                  +new

Summary for example
------------
Succeeded: 1 (unchanged=1, changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  54.081 ms
```

`file.managed` using a template:

```
$ cat states/test.sls 
/tmp/test:
  file.managed:
    - source: salt://test.j2
    - template: jinja

$ cat states/test.j2
new

$ salt-ssh example state.apply test test=true
example:
----------
          ID: /tmp/test
    Function: file.managed
      Result: None
     Comment: The file /tmp/test is set to be changed
     Started: 20:25:35.639444
    Duration: 58.472 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -1 +1 @@
                  -new
                  +old

Summary for example
------------
Succeeded: 1 (unchanged=1, changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  58.472 ms
```

```
Salt Version:
           Salt: 2018.3.0-290-g8105fd9
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.3 (default, Jan 19 2017, 14:11:04)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.6
 
System Versions:
           dist: debian 9.4 
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-6-amd64
         system: Linux
        version: debian 9.4
```

**Comment by Ch3LL** [comment]
Looks like i'm able to replicate this. I'm having a hard time bisecting this to find the first commit that broke this though. ping @terminalmage any ideas here?

To note when removing test=True and running the state it does work and pastes the correct order.

**Comment by gtmanfred** [comment]
This will be fixed in #47046 Which will be in 2018.3.1

Thanks!
Daniel
